File: app/src/main/java/me/destro/android/gitfav/features/listing/StarredRepositoriesFragment.java
package me.destro.android.gitfav.features.listing;

import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;
import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.paging.PagedList;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import me.destro.android.gitfav.R;
import me.destro.android.gitfav.databinding.FragmentStarredRepositoriesBinding;
import me.destro.android.gitfav.features.listing.adapters.StarredRepositoriesAdapter;
import me.destro.android.libraries.github.model.StarredRepository;


public class StarredRepositoriesFragment extends Fragment {

    private FragmentStarredRepositoriesBinding binding;

    private RecyclerView rvStarredRepositories;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_starred_repositories, container, false);

        rvStarredRepositories = binding.starredRepositories;

        String username = StarredRepositoriesFragmentArgs.fromBundle(getArguments()).getUsername();

        StarredRepositoriesAdapter mAdapter = new StarredRepositoriesAdapter();
        mAdapter.setOnStarredRepositoryClickListener((starredRepository) -> {
            StarredRepositoriesFragmentDirections.ActionStarredRepositoriesFragmentToRepositoryDetailFragment
                    action = StarredRepositoriesFragmentDirections.actionStarredRepositoriesFragmentToRepositoryDetailFragment(starredRepository.name, starredRepository.owner.getLogin());

            NavController navigation = Navigation.findNavController(binding.getRoot());
            navigation.navigate(action);
        });
        rvStarredRepositories.setAdapter(mAdapter);
        rvStarredRepositories.setLayoutManager(new LinearLayoutManager(getContext()));

        StarredRepositoriesViewModel viewModel = ViewModelProviders.of(this).get(StarredRepositoriesViewModel.class);
        viewModel.getStarredRepositories(username).observe(this, new Observer<PagedList<StarredRepository>>() {
            @Override
            public void onChanged(@Nullable PagedList<StarredRepository> starredRepositories) {
                mAdapter.submitList(starredRepositories);
            }
        });

        return binding.getRoot();
    }
}

File: app/src/main/java/me/destro/android/gitfav/features/listing/StarredRepositoriesFragment.kt
package me.destro.android.gitfav.features.listing

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.navigation.Navigation
import androidx.paging.PagedList
import androidx.recyclerview.widget.LinearLayoutManager
import me.destro.android.gitfav.R
import me.destro.android.gitfav.databinding.FragmentStarredRepositoriesBinding
import me.destro.android.gitfav.domain.model.Repository
import me.destro.android.gitfav.features.listing.adapters.StarredRepositoriesAdapter
import org.koin.androidx.viewmodel.ext.android.viewModel

class StarredRepositoriesFragment : Fragment() {

    private lateinit var binding: FragmentStarredRepositoriesBinding

    private val viewModel by viewModel<StarredRepositoriesViewModel>()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_starred_repositories, container, false)

        val rvStarredRepositories = binding.starredRepositories

        val username = StarredRepositoriesFragmentArgs.fromBundle(arguments!!).username

        val mAdapter = StarredRepositoriesAdapter()
        mAdapter.setOnStarredRepositoryClickListener { repository ->
            val action = StarredRepositoriesFragmentDirections.actionStarredRepositoriesFragmentToRepositoryDetailFragment(repository.name, repository.owner)

            val navigation = Navigation.findNavController(binding.root)
            navigation.navigate(action)
        }

        rvStarredRepositories.adapter = mAdapter
        rvStarredRepositories.layoutManager = LinearLayoutManager(context)

        viewModel.getStarredRepositories(username).observe(this, Observer<PagedList<Repository>> { starredRepositories -> mAdapter.submitList(starredRepositories) })

        return binding!!.root
    }
}
